en/uavcan/node_enumeration.md

```diff
@@ -17,3 +17,4 @@ ESCs running the Sapog firmware will need to be rebooted after enumeration for t
 This step has to be performed only once and does not need to be repeated after firmware upgrades.
```

Comment:
@bkueng I think this doc essentially is a copy of the docs we now have in the esc setup.

What I think needs to happen is that this goes into the parent config doc, which explains there are the three options set by UAVCAN_ENABLE. My assumption
1: Sensors Manual Config - if you set this then you have to configure everything manually using "native" config tools. You probably need this for anything other than Sapog right?
2: Sensors Automatic Config - This basically enumerates SAPOG (only) sensors - right?
3: Sensors and Actuators (ESCs) Automatic Config - This is the ESC config. You set this then you can enumerate the motors.

I have a question though, what does the control highlighted in red below do? (not clickable on my computer)
![image](https://user-images.githubusercontent.com/5368500/109113024-a9ef6800-778f-11eb-98ad-74419fb0e1e9.png)

Reply:
1. This is if all your devices have a fixed node ID - you need to ensure there's no duplicate
2. Any device might support dynamic enumeration, it needs to be checked in the manual
3. yes

> I have a question though, what does the control highlighted in red below do?

No idea

Reply:
@bkueng Sorry to come back to this. Some more checks on assumptions, in particular when you use the UI and the different options for `UAVCAN_ENABLE`.

1. `UAVCAN_ENABLE=1` means manual enumeration of ids. Am I correct in thinking that you can always set this, but then
   - you will need to manually enumerate all sensors and ESCs? 
   - You would use whatever tools are appropriate/supplied for your ESC and other nodes - e.g. VESC or Sapog or whatever.
2. `UAVCAN_ENABLE=2` means dynamic enumeration. I.e. PX4 will try an enumerate all nodes EXCEPT for your ESCs.
   - Is that correct? I.e. this would enumerate your GNSS, power module, whatever, irrespective of type?
   - What if some of your devices do not support dynamic enumeration? Would you set to this and assignment will happen automatically for those that support it, and then do the others manually?
   - How does the enumeration happen - is it automatic on every boot, or is it a one off and then saved? Do you need to do anything else in the UI or is just setting the value enough?
3. `UAVCAN_ENABLE=3` means dynamic enumeration of sensors and automatic enumeration of ESCS.
   - Does this work on all ESC or just Sapog?
   - What if you have a sensor that doesn't support dynamic allocation - do you use this, do the assignment then manually assign the static allocation using the type-specific tools?
   - If it doesn't work on all ESC, then what do you do for VESC say? Select 2 to do the sensors and then use the VESC tools to assign the ESC.

Hope the question makes sense!

Reply:
These are all good questions, and definitely worth clarifying in detail. But you may have to ask someone else for more details.

> you will need to manually enumerate all sensors and ESCs?

Yes

> You would use whatever tools are appropriate/supplied for your ESC and other nodes - e.g. VESC or Sapog or whatever.

Yes, essentially statically assign them

> PX4 will try an enumerate all nodes EXCEPT for your ESCs.

Presumably, but I'm not 100% sure on the exact details

> What if some of your devices do not support dynamic enumeration?

Good question. :shrug: Probably manual assignment but using a high enough ID to ensure no overlap.

> How does the enumeration happen - is it automatic on every boot, or is it a one off and then saved?

There's a database stored to SD card to ensure it stays consistent.

> Does this work on all ESC or just Sapog?

I don't know, I think it needs special support.

> What if you have a sensor that doesn't support dynamic allocation - do you use this, do the assignment then manually assign the static allocation using the type-specific tools?

I guess so

> If it doesn't work on all ESC, then what do you do for VESC say? Select 2 to do the sensors and then use the VESC tools to assign the ESC.

I don't think you would mix ESCs in the first place, but yes, that could work.